crates/aquascope/src/analysis/boundaries/mod.rs

```diff
@@ -71,6 +71,72 @@ impl PermissionsBoundary {
 // ----------------------------------
 // Permission boundaries on path uses
 
+#[derive(Copy, Clone, Debug)]
+struct ExpectedPermissions(Permissions);
+
+impl ExpectedPermissions {
+  pub fn from_receiver_ty<'tcx>(
```

Comment:
Something we should document, either here or in the paper, is that receiver types are special because Rust permits more coercions for receivers than for other function arguments.